mm2src/coins/z_coin/storage/walletdb/wasm/storage.rs

```diff
@@ -1248,22 +1248,24 @@ impl WalletRead for WalletIndexedDb {
 
         let mut nullifiers = vec![];
         for (_, note) in maybe_notes {
-            let matching_tx = maybe_txs.iter().find(|(id_tx, _tx)| id_tx.to_bigint() == note.spent);
-
-            if let Some((_, tx)) = matching_tx {
-                if tx.block.is_some() {
-                    nullifiers.push((
-                        AccountId(
-                            note.account
-                                .to_u32()
-                                .ok_or_else(|| ZcoinStorageError::GetFromStorageError("Invalid amount".to_string()))?,
-                        ),
-                        Nullifier::from_slice(&note.nf.clone().ok_or_else(|| {
-                            ZcoinStorageError::GetFromStorageError("Error while putting tx_meta".to_string())
-                        })?)
-                        .unwrap(),
-                    ));
-                }
+            let maybe_spending_tx = maybe_txs.iter().find(|(id_tx, _tx)| id_tx.to_bigint() == note.spent);
+            let add_nullifier = match maybe_spending_tx {
+                Some((_, tx)) if tx.block.is_none() => true,
+                None => true,
+                _ => false,
+            };
+            if add_nullifier {
+                nullifiers.push((
+                    AccountId(
+                        note.account
+                            .to_u32()
+                            .ok_or_else(|| ZcoinStorageError::GetFromStorageError("Invalid amount".to_string()))?,
+                    ),
+                    Nullifier::from_slice(&note.nf.clone().ok_or_else(|| {
+                        ZcoinStorageError::GetFromStorageError("Error while putting tx_meta".to_string())
+                    })?)
+                    .unwrap(),
+                ));
```

Comment:
I would avoid unwrap, or enforce `nf` to be `NOT NULL` like native.
```suggestion
                if let Some(nf_bytes) = note.nf.clone() {
                    let account_id = AccountId(
                        note.account
                            .to_u32()
                            .ok_or_else(|| ZcoinStorageError::GetFromStorageError("Invalid account id".to_string()))?,
                    );
                    let nf = Nullifier::from_slice(&nf_bytes).map_err(|e| {
                        ZcoinStorageError::GetFromStorageError(format!("Invalid nullifier bytes error: {}", e))
                    })?;
                    nullifiers.push((account_id, nf));
                }
```

Reply:
Fixed as you suggested, thanks https://github.com/KomodoPlatform/komodo-defi-framework/pull/2651/commits/350b1b2dba25fe655c27f19adca4b2e11e54b43d. 
I also thought we should skip maybe-null nullifiers although they are not possible in the current codebase.

(Also thank you for finding the original cause of this issue)